bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/inbox/events/InboxEventFactory.java

```diff
@@ -20,7 +20,6 @@
 import org.eclipse.smarthome.core.events.EventFactory;
 import org.osgi.service.component.annotations.Component;
 
-import com.google.common.base.Preconditions;
 import com.google.common.collect.Sets;
```

Comment:
why not replace Sets usage in the constructor?

```
public InboxEventFactory() {
    super(new HashSet<>(Arrays.asList(InboxAddedEvent.TYPE, InboxUpdatedEvent.TYPE, InboxRemovedEvent.TYPE)));
}
```

Reply:
because it technically would even have to be 

```java
super(Collections.unmodifiableSet(new HashSet<>(Arrays.asList(InboxAddedEvent.TYPE, InboxUpdatedEvent.TYPE, InboxRemovedEvent.TYPE))))
```

I'd rather want to have a convenience method which does exactly this (maybe a little optimized), but don't neither clutters the code as much nor drags in a couple of MBs footprint like guava. Nevertheless, I left it out of the scope if _this_ PR to keep it manageable.

Reply:
okay, so this be part of the second or third run then.

Reply:
Why not using something like `Stream.of(InboxAddedEvent.TYPE, InboxUpdatedEvent.TYPE, InboxRemovedEvent.TYPE).collect(toSet())`?